hash/hasher_test.go

```diff
@@ -122,30 +137,97 @@ func TestComparatorBcryptFail(t *testing.T) {
 		mkpw(t, 72),
 	} {
 		t.Run(fmt.Sprintf("case=%d", k), func(t *testing.T) {
+			_, reg := internal.NewFastRegistryWithMocks(t)
 			mod := make([]byte, len(pw))
 			copy(mod, pw)
 			mod[len(pw)-1] = ^pw[len(pw)-1]
 
-			err := hash.CompareBcrypt(context.Background(), pw, mod)
+			err := hash.CompareBcrypt(context.Background(), reg.Config(context.Background()), pw, string(mod))
+			assert.Error(t, err)
+		})
+	}
+}
+
+func TestComparatorBcryptAesSuccess(t *testing.T) {
+	for k, pw := range [][]byte{
+		mkpw(t, 8),
+		mkpw(t, 16),
+		mkpw(t, 32),
+		mkpw(t, 64),
+		mkpw(t, 72),
+		mkpw(t, 96),
+		mkpw(t, 128),
+	} {
+		t.Run(fmt.Sprintf("case=%d", k), func(t *testing.T) {
+			_, reg := internal.NewFastRegistryWithMocks(t)
+			hasher := hash.NewHasherBcryptAES(reg)
+
+			hs, err := hasher.Generate(context.Background(), pw)
+
+			assert.Nil(t, err)
+
+			algorithm, realHash, err := hash.ParsePasswordHash(hs)
+			assert.NoError(t, err)
+			assert.Equal(t, hash.BcryptAESAlgorithmId, algorithm)
+
+			hs2, _ := hasher.Generate(context.Background(), []byte("test"))
+			fmt.Printf("test: %s\n", string(hs2))
+
+			err = hash.CompareBcryptAes(context.Background(), reg.Config(context.Background()), pw, realHash)
+			assert.Nil(t, err, "hash validation fails")
+		})
+	}
+}
+
+func TestComparatorBcryptAesFail(t *testing.T) {
+	p := config.MustNew(t, logrusx.New("", ""),
+		configx.WithConfigFiles("../internal/.kratos.yaml"))
+
+	for k, pw := range [][]byte{
+		mkpw(t, 8),
+		mkpw(t, 16),
+		mkpw(t, 32),
+		mkpw(t, 64),
+		mkpw(t, 72),
+		mkpw(t, 96),
+		mkpw(t, 128),
+	} {
+		t.Run(fmt.Sprintf("case=%d", k), func(t *testing.T) {
+			mod := make([]byte, len(pw))
+			copy(mod, pw)
+			mod[len(pw)-1] = ^pw[len(pw)-1]
```

Comment:
Just to make it obvious
```suggestion
			mod[len(pw)-1] = ^mod[len(pw)-1]
```